api/src/catalog/sitemap.utils.ts

```diff
@@ -0,0 +1,66 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { Readable } from 'node:stream';
+import { SitemapStream, streamToPromise } from 'sitemap';
+import { getSdk, type CatalogBySeasonQuery } from './catalog.queries.js';
+import { STATIC_FILE_DIR, graphqlClient } from '../config.js';
+import winston from '../logging/winston.js';
+
+interface Course {
+  crn: string;
+}
+
+function transformListingToCourse(
+  listing: CatalogBySeasonQuery['listings'][number],
+): Course {
+  return {
+    crn: String(listing.crn),
+  };
+}
+
+async function generateSeasonSitemap(
+  seasonCode: string,
+  courses: Course[],
+): Promise<void> {
+  const sitemapDir = path.join(STATIC_FILE_DIR, 'sitemaps');
+  await fs.mkdir(sitemapDir, { recursive: true });
+
+  const links = courses.map((course) => ({
+    url: `/catalog?course-modal=${seasonCode}-${course.crn}`,
+    priority: 0.8,
+  }));
+
+  const stream = new SitemapStream({ hostname: 'https://coursetable.com' });
+  const xml = await streamToPromise(Readable.from(links).pipe(stream)).then(
+    (data: Buffer) => data.toString(),
+  );
+
+  const sitemapPath = path.join(sitemapDir, `sitemap_${seasonCode}.xml`);
+  await fs.writeFile(sitemapPath, xml, 'utf-8');
+
+  winston.info(`Sitemap generated for ${seasonCode} at ${sitemapPath}`);
+}
+
+export async function generateSitemaps(): Promise<void> {
+  let seasons: string[] = [];
+  try {
+    const sdk = getSdk(graphqlClient);
+    seasons = (await sdk.listSeasons()).seasons.map((x) => x.season_code);
```

Comment:
No need to fetch twice—the `fetchCatalog` function already did both of these fetches. You can either do this function as part of `fetchCatalog`, or let `fetchCatalog` return the data you need, or even just read the JSON files that `fetchCatalog` wrote—your call.